pkg/cosmos/client/gas_price_estimator.go

```diff
@@ -17,16 +19,77 @@ var _ GasPricesEstimator = (*FixedGasPriceEstimator)(nil)
 
 type FixedGasPriceEstimator struct {
 	gasPrices map[string]sdk.DecCoin
+	lggr      logger.SugaredLogger
 }
 
-func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin) *FixedGasPriceEstimator {
-	return &FixedGasPriceEstimator{gasPrices: prices}
+func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin, lggr logger.SugaredLogger) *FixedGasPriceEstimator {
+	return &FixedGasPriceEstimator{gasPrices: prices, lggr: lggr}
 }
 
 func (gpe *FixedGasPriceEstimator) GasPrices() (map[string]sdk.DecCoin, error) {
 	return gpe.gasPrices, nil
 }
 
+func (gpe *FixedGasPriceEstimator) GasPrice(coin string) sdk.DecCoin {
+	return gpe.gasPrices[coin]
+}
+
+func (gpe *FixedGasPriceEstimator) SetGasPrice(coin string, price sdk.DecCoin) {
+	gpe.gasPrices[coin] = price
+}
+
+// CalculateGasPrice calculates the minimum of the maximum input gas prices.
+// Parameters:
+// - maxGasPrice: max gas price set by the user or operator
+// - defaultGasPrice: default gas price
+// - maxGasPriceConfigured: max gas price originally configured for the node
+func (gpe *FixedGasPriceEstimator) CalculateGasPrice(
+	coin string,
+	maxGasPrice,
+	defaultGasPrice,
+	maxGasPriceConfigured sdk.DecCoin,
+) sdk.DecCoin {
+	calculatedGasPrice := fee.CalculateFee(
+		maxGasPrice.Amount.BigInt(),
+		defaultGasPrice.Amount.BigInt(),
+		maxGasPriceConfigured.Amount.BigInt(),
+	)
+	return sdk.NewDecCoinFromDec(coin, sdk.NewDecFromBigIntWithPrec(calculatedGasPrice, 18))
+}
+
+// BumpGasPrice calculates a new gas price by bumping the current gas price by a percentage.
+// Parameters:
+// - currentGasPrice: current gas price (could be after bumping)
+// - originalGasPrice: original gas price (before bumping)
+// - maxGasPrice: max gas price
+// - maxBumpPrice: max gas price that can be bumped to
```

Comment:
I know that we are just passing these straight to fee.CalculateBumpedFee, but just curious if there is a use case where these values should differ?

Reply:
Thanks for clarifying this calvin. 
```
// - maxGasPrice: max gas price
// - maxBumpPrice: max gas price that can be bumped to
```
Logically they are the same where they are a cap on the gas price we are willing to pay. 

Usecase wise, i would think that they can be different in the sense where 
- Max bump price can be a default value set across the node. Max gas price can be an input that can be raised or decreased. This is how EVM is doing it now. 

Do you think we should use the same value for both?

Reply:
Currently, it seems to me that there are quite a lot of inputs and context required to use this bumped gas fee calculator, which could possibly be broken down into helper functions or calculated outside of the function and handled by the consumer. These are more suggestions for the generic fee estimator interface rather than this cosmos PR so these suggestions are not actionable here nor blocking this PR 👍 

we need to provide both:
- current gas price of the network
- original gas price
which is aggregated into a base gas price by taking the max. Having instead one input of `base_gas_price` means that less context needs to be shared with this helper function, the input/outputs are clearer to the consumer, and the consumer has the flexibility to choose which base to use. godoc documentation and/or helper fns can detail the recommended approach which is `max(current, original)`.

we also need to provide:
- min gas price bump
- max gas price bump
- max gas price overall
- % to bump gas price
essentially defining a bounded range (min to max) and a target (% of base) for the new gas price. context outside of this information (i.e. max bump vs max overall) is tangential to the calculation itself, and by aggregating it into one function input we can improve the clarity of the function inputs/outputs and follow a more single-responsibility approach.

Reply:
Thanks for the feedback Calvin, appreciate the breakdown. Let me see if I can improve this by abstracting away the function parameters.